Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/GenerateKeyFlow.swift

```diff
@@ -35,37 +34,40 @@ class GenerateKeyFlow: AddKeyFlow {
     var addOwnerFlow: AddOwnerFlow!
     var replaceOwnerFlow: ReplaceOwnerFlow!
 
+    var parameters: GenerateKeyParameters? {
+        keyParameters as? GenerateKeyParameters
+    }
+
     init(completion: @escaping (Bool) -> Void) {
-        super.init(keyType: .deviceGenerated, factory: GenerateKeyFactory(), completion: completion)
+        super.init(factory: GenerateKeyFactory(), completion: completion)
     }
 
     override func didIntro() {
         let privateKey = OwnerKeyController.generate()
-        didGetKey(privateKey: privateKey)
+        keyParameters = GenerateKeyParameters(address: privateKey.address, keyName: nil, privateKey: privateKey)
+        didGetKey()
     }
 
     override func doImport() -> Bool {
-        assert(privateKey != nil)
-        assert(keyName != nil)
-        let success = OwnerKeyController.importKey(privateKey!, name: keyName!, isDrivedFromSeedPhrase: true)
+        assert(parameters?.name != nil)
+        assert(parameters?.privateKey != nil)
+        let success = OwnerKeyController.importKey(parameters!.privateKey!,
+                                                   name: parameters!.name!,
+                                                   isDrivedFromSeedPhrase: true)
         return success
     }
 
-    override func didCreatePasscode() {
-        backup()
-    }
-
     func backup() {
-        assert(privateKey?.mnemonic != nil)
-        backupFlow = BackupFlow(mnemonic: privateKey!.mnemonic!) { [unowned self] _ in
+        assert(parameters?.privateKey?.mnemonic != nil)
+        backupFlow = BackupFlow(mnemonic: parameters!.privateKey!.mnemonic!) { [unowned self] _ in
```

Comment:
so many forced unwraps here and in other places

Reply:
all are protected with the assert before right?, how do you suggest to handle it ?

Reply:
well... what options do we have and  can we try out and see what works better?
* guard let
* making struct fields forced unwrapped

Reply:
@MouazAlzahabi is it only in this place or in other functions and flows the same thing happens?

Reply:
So as parameters are optional and some of its properties are optional then you might find force unwrapping in multiple places, but all are protected with check before

Reply:
what I'm trying to say that this problem is not only in this place but in multiple places, so the fix should be applied consistently. Please check out if the guard-let or forced unwrapping... basically so many ! in the code itself making it hard to reason / read, because then I question: why so many forced unwraps? is it intended?

Reply:
@DmitryBespalov I did improvement for other places wdyt?